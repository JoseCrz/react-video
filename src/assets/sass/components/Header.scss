.Header {
  width: 100%;
  min-height: 80px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  &__logo {
    margin-top: 10px;
    margin-left: 10px;
  }

  &__nav-profile {
    /* ? Solo así funcionó en Safari... */
    display: -webkit-box;
    display: -ms-flexbox;
    /* display: flex; */
    margin-right: 10px;
  }

  &__nav-profile img {
    margin-right: 8px;
  }

  &__nav-menu {
    display: none;
    list-style: none;
    padding: 0;
    margin: 0;
    margin-right: -8px;
    position: absolute;
    width: 100px;
    text-align: right;
  }

  &__nav-menu li {
    margin: 10px 0;
  }

  &__nav-menu a {
    color: white;
    text-decoration: none;
    margin-right: -5px;

    &:hover {
      text-decoration: underline;
    }
  }

  &__nav:hover &__nav-menu, 
  &__nav-menu:hover {
    display: block;
  }
}
