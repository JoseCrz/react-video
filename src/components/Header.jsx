import React from 'react'

import '../assets/sass/components/Header.scss'

const Header = () => {
  return (
    <header className="Header">
        <figure className="Header__logo-container">
            <img className="Header__logo" src="./img/logo-white.png" alt="Reactivid Logo" width="150px" />
        </figure>
        <nav className="Header__nav">
            <div className="Header__nav-profile">
                <img src="./img/user-icon.svg" alt="Profile Icon" width="50px" />
                <p>Profile</p>
            </div>
            <ul className="Header__nav-menu">
                <li><a href="#">Settings</a></li>
                <li><a href="#">Sign out</a></li>
            </ul>
        </nav>
    </header>
  )
}

export default Header