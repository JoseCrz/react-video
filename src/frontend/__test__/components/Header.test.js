import React from 'react'
import { mount }  from 'enzyme'
import { create } from 'react-test-renderer'
import Header from '../../components/Header'
import ProviderMock from '../../__mocks__/providerMock'


describe('<Header /> Tests', () => {
  
  const header = mount(
    <ProviderMock>
      <Header />
    </ProviderMock>
  )

  test('Component rendering', () => {
    expect(header.length).toEqual(1)
  })

  test('Logo existance', () => {
    expect(header.find('.Header__logo')).toHaveLength(1)
  })

  test('Component snapshot', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>
    )

    expect(header.toJSON()).toMatchSnapshot()
  })

})