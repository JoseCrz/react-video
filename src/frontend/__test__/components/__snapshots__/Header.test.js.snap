// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Tests Component snapshot 1`] = `
<header
  className="Header"
>
  <figure
    className="Header__logo-container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Reactivid Logo"
        className="Header__logo"
        src="test-file-stub"
      />
    </a>
  </figure>
  <nav
    className="Header__nav"
  >
    <div
      className="Header__nav-profile"
    >
      <img
        alt="Profile Icon"
        className="Header__profile"
        src="test-file-stub"
      />
      <p>
        Profile
      </p>
    </div>
    <ul
      className="Header__nav-menu"
    >
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
